plugins/sqlcipherplugin/sqlcipherplugin.h

```diff
@@ -57,8 +59,8 @@ class CryptoPlugin : public Sailfish::Crypto::CryptoPlugin
 };
 
 class Q_DECL_EXPORT SqlCipherPlugin : public QObject
-                                    , public Sailfish::Secrets::Daemon::Plugins::EncryptedStoragePlugin
-                                    , public Sailfish::Secrets::Daemon::Plugins::CryptoPlugin
+                                    , public virtual Sailfish::Secrets::Daemon::Plugins::EncryptedStoragePlugin
```

Comment:
Because the SqlCipher plugin derives from both CryptoPlugin and EncryptedStoragePlugin, and this commit refactors the Plugin types to derive from a common base class (as per your and Venemo's review feedback), thus we have diamond multiple inheritance.  The most-base class is ABC but nonetheless virtual inheritance ensures we have only a single base instance.